src/features/smartSearch/components/filters/AllInSubOrg/index.tsx

```diff
@@ -0,0 +1,223 @@
+  SmartSearchFilterWithId,
+  ZetkinSmartSearchFilter,
+} from '../../types';
+import StyledSelect from '../../inputs/StyledSelect';
+import { Msg } from 'core/i18n';
+import messageIds from 'features/smartSearch/l10n/messageIds';
+import useSubOrganizations from 'features/organizations/hooks/useSubOrganizations';
+import { useNumericRouteParams } from 'core/hooks';
+import useSmartSearchFilter from 'features/smartSearch/hooks/useSmartSearchFilter';
+import StyledItemSelect from '../../inputs/StyledItemSelect';
+
+type Props = {
+  filter:
+    | SmartSearchFilterWithId<AllInSuborgFilterConfig>
+    | NewSmartSearchFilter;
+  onCancel: () => void;
+  onSubmit: (
+    filter:
+      | SmartSearchFilterWithId<AllInSuborgFilterConfig>
+      | ZetkinSmartSearchFilter<AllInSuborgFilterConfig>
+  ) => void;
+};
+
+const AllInSuborg: FC<Props> = ({
+  filter: initialFilter,
+  onSubmit,
+  onCancel,
+}) => {
+  const { orgId } = useNumericRouteParams();
+  const suborgs = useSubOrganizations(orgId).data || [];
+  const [scope, setScope] = useState<'any' | 'single' | 'multiple'>('any');
+  const [selectedSuborgIds, setSelectedSuborgIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    setSelectedSuborgIds([]);
+  }, [scope]);
+
+  const { filter, setConfig, setOp } =
+    useSmartSearchFilter<AllInSuborgFilterConfig>(initialFilter, {
+      organizations: 'suborgs',
+    });
+
+  const activeSuborgs = suborgs
+    .filter((suborg) => suborg.id != orgId)
+    .filter((suborg) => suborg.is_active)
+    .sort((a, b) => a.title.localeCompare(b.title));
+
+  const addRemoveSelect = (
+    <StyledSelect
+      onChange={(e) => setOp(e.target.value as OPERATION)}
+      value={filter.op}
+    >
+      {Object.values(OPERATION).map((o) => (
+        <MenuItem key={o} value={o}>
+          <Msg id={messageIds.operators[o]} />
+        </MenuItem>
+      ))}
+    </StyledSelect>
+  );
+
+  const suborgScopeSelect = (
+    <StyledSelect
+      onChange={(e) => {
+        const newValue = e.target.value as 'any' | 'single' | 'multiple';
+        setScope(newValue);
+
+        if (newValue == 'any') {
+          setConfig({
+            ...filter.config,
+            organizations: 'suborgs',
+          });
+        }
+      }}
+      value={scope}
+    >
+      <MenuItem value="any">
+        <Msg id={messageIds.filters.allInSuborg.suborgScopeSelect.any} />
+      </MenuItem>
+      <MenuItem value="single">
+        <Msg id={messageIds.filters.allInSuborg.suborgScopeSelect.single} />
+      </MenuItem>
+      <MenuItem value="multiple">
+        <Msg id={messageIds.filters.allInSuborg.suborgScopeSelect.multiple} />
+      </MenuItem>
+    </StyledSelect>
+  );
+
+  const singleSuborgSelect = (
+    <StyledSelect
+      onChange={(ev) => {
+        const selectedSuborgId = parseInt(ev.target.value);
+        setSelectedSuborgIds([selectedSuborgId]);
+        setConfig({
+          ...filter.config,
+          organizations: [selectedSuborgId],
+        });
```

Comment:
Is it necessary to do both and even have a local state, if the local state (`setSelectedSuborgIds`) and the config state (`setConfig`) match 1:1?

Reply:
there was at some point a reason, but i will investigate again if that reason still stands!